src/main/java/edu/ucsb/ucsbcslas/controllers/OnlineOfficeHoursController.java

```diff
@@ -120,4 +134,100 @@
+           return getUnauthorizedResponse("admin");
+        try {
+            BufferedReader csvReader = new BufferedReader(new InputStreamReader(csv.getInputStream()));
+            logger.info(new String(csv.getInputStream().readAllBytes()));
+            
+            String body = "";
+
+            List<OnlineOfficeHours> savedOfficeHour = new ArrayList<>();
+
+            String row = csvReader.readLine();
+            String [] data ;
+            while(row != null){
+                row = row.replace("\"","");
+                data = row.split(",");
+
+
+                String name = data[0];
+                String quarter = data[1];
+                String instructorFirstName = data[2];
+                String instructorLastName = data[3];
+                String instructorEmail = data[4];
+                String firstName = data[5];
+                String lastName = data[6];
+                String email = data[7];
+                String assignmentType = data[8];
+                String dayOfWeek = data[9];
+                String startTime = data[10];
+                String endTime = data[11];
+                String zoomRoomLink = data[12];
+                String notes = data[13];
+                
+                row = csvReader.readLine();
+                
+                Course course = new Course(name,quarter,instructorFirstName,instructorLastName,instructorEmail);
+                
+                Course existingCourse = courseRepository.findByNameAndQuarter(course.getName(), course.getQuarter());
+                if(existingCourse == null){
+                    courseRepository.save(course);
+                }
+                
+
+                Tutor tutor = new Tutor(firstName,lastName,email);
+                Tutor related_tutor;
+                Optional<Tutor> existingTutor = tutorRepository.findByEmail(email);
+                
+                Course related_course = courseRepository.findByNameAndQuarter(course.getName(), course.getQuarter());
+
+                if(existingTutor.isEmpty()){
+                    related_tutor = tutorRepository.save(tutor);
+                }
+                else{
+                    related_tutor = existingTutor.get();
+                }
+
+                
+                TutorAssignment TA;
+                TutorAssignment tutorAssignment = new TutorAssignment(related_course,related_tutor,assignmentType);
+                List<TutorAssignment> TAbyCourse = tutorAssignmentRepository.findAllByCourse(related_course);
+                List<TutorAssignment> TAbyTutor = tutorAssignmentRepository.findAllByTutor(related_tutor);
+                List<TutorAssignment> comm =TAbyCourse;
+                comm.retainAll(TAbyTutor);
+                if(comm.isEmpty()){
+                    TA = tutorAssignmentRepository.save(tutorAssignment);
+                }
+                else{
+                    TA = comm.get(0);
+                }
+                
```

Comment:
It should be possible and relatively straightforward to replace this chunk of logic with a single database query. Rather than calling `findAllByCourse` and `findAllByTutor` and then finding the intersection manually, try adding a new query to the TutorAssignmentRepository to `findAllByCourseAndTutor` and using that instead here

Reply:
Added and changed.